frontend/apps/filemanager/filemanager.lua

```diff
@@ -436,8 +443,8 @@ function FileManager:setupLayout()
         end
 
         local title
-        if lfs.attributes(file, "mode") == "directory" then
-            title = BD.directory(file:match("([^/]+)$"))
+        if is_folder then
+            title = BD.directory(file:match("([^/]+)$")).."/"
```

Comment:
Not super sure about it, but it feels like the trailing / should be part of what will be BD.directory()'ed (some BiDi unicode control chars so it gets ok in BiDi text).

Reply:
> the trailing / should be part of what will be BD.directory()'ed

I don't have it on Kindle.
If it is on other systems, my addition should be removed for sure.

Reply:
You don't have what ? :)
It's something that you might only see in an Arabic/Hebrew UI, or in a LTR UI with maybe files and directories in arabic or hebrew.
I don't remember much about all that, but it's probably less risky to have it output what would be happening here:
https://github.com/koreader/koreader/blob/de543b348468820aef5015285e35cd740ea2a4e1/frontend/ui/widget/filechooser.lua#L281-L287

So, in your case, maybe:
```lua
        if is_folder then
            local dir_name = file:match("([^/]+)$") -- I'm not sure this might not returns multiple values, so keep the ../ for the next one
            title = BD.directory(dir_name .. "/")
```

Reply:
> `-- will keep any trailing / on the right`

That's the point: with your current code, in an Arabic UI, if the directory/title is `/koreader`, and you add the `/` after BD.directory(), it may get displayed as : `//koreader`.
If you add the / inside BD.directory(), you should get the expected `/koreader/`

Reply:
Thank you, this bidi thing requires more learning (for me), leave it for future.